source/backend/dal/Repositories/ExpropriationPaymentRepository.cs

```diff
@@ -0,0 +1,90 @@
+        }
+
+        public IList<PimsExpropriationPayment> GetAllByAcquisitionFileId(long acquisitionFileId)
+        {
+            return Context.PimsExpropriationPayments
+                .Include(x => x.PimsExpropPmtPmtItems)
+                    .ThenInclude(y => y.PaymentItemTypeCodeNavigation)
+                .Include(x => x.InterestHolder)
+                    .ThenInclude(y => y.Person)
+                .Include(x => x.InterestHolder)
+                    .ThenInclude(y => y.Organization)
+                .Include(x => x.AcquisitionOwner)
+                .Include(x => x.ExpropriatingAuthorityNavigation)
+                .AsNoTracking()
+                .Where(x => x.AcquisitionFileId == acquisitionFileId)
+                .ToList();
+        }
+
+        public PimsExpropriationPayment GetById(long expropriationPaymentId)
+        {
+            return Context.PimsExpropriationPayments
+                .Include(x => x.PimsExpropPmtPmtItems)
+                    .ThenInclude(y => y.PaymentItemTypeCodeNavigation)
+                .Include(x => x.InterestHolder)
+                    .ThenInclude(y => y.Person)
+                .Include(x => x.InterestHolder)
+                    .ThenInclude(y => y.Organization)
+                .Include(x => x.AcquisitionOwner)
+                .Include(x => x.ExpropriatingAuthorityNavigation)
+                .AsNoTracking()
+                .FirstOrDefault(x => x.ExpropriationPaymentId.Equals(expropriationPaymentId)) ?? throw new KeyNotFoundException();
+        }
+
+        public PimsExpropriationPayment Udpate(PimsExpropriationPayment expropriationPayment)
+        {
+            var existingExpPayment = Context.PimsExpropriationPayments
+                .FirstOrDefault(x => x.ExpropriationPaymentId.Equals(expropriationPayment.ExpropriationPaymentId)) ?? throw new KeyNotFoundException();
+
+            Context.Entry(existingExpPayment).CurrentValues.SetValues(expropriationPayment);
+            Context.UpdateChild<PimsExpropriationPayment, long, PimsExpropPmtPmtItem, long>(a => a.PimsExpropPmtPmtItems, expropriationPayment.ExpropriationPaymentId, expropriationPayment.PimsExpropPmtPmtItems.ToArray(), true);
+
+            return existingExpPayment;
+        }
+
+        public bool TryDelete(long id)
+        {
+            var deletedEntity = Context.PimsExpropriationPayments
+                .Include(x => x.PimsExpropPmtPmtItems)
+                .AsNoTracking()
+                .FirstOrDefault(c => c.ExpropriationPaymentId == id);
+
+            if (deletedEntity != null)
+            {
+                foreach (var item in deletedEntity.PimsExpropPmtPmtItems)
+                {
+                    Context.PimsExpropPmtPmtItems.Remove(new PimsExpropPmtPmtItem() { ExpropPmtPmtItemId = item.ExpropPmtPmtItemId });
+                }
+
+                Context.CommitTransaction(); // TODO: required to enforce delete order. Can be removed when cascade deletes are implemented.
```

Comment:
Oh doug, you will be the end of me I swear.

Reply:
would be easier if we configure the cascades with EF

Reply:
we can't do my knowledge, since we are db first.